test/Chainweb/Test/Pact/RemotePactTest.hs

```diff
@@ -0,0 +1,272 @@
+{-# LANGUAGE MultiWayIf #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+-- |
+-- Module: Chainweb.Test.RemotePactTest
+-- Copyright: Copyright © 2019 Kadena LLC.
+-- License: See LICENSE file
+-- Maintainer: Mark Nichols <mark@kadena.io>
+-- Stability: experimental
+--
+-- Unit test for Pact execution via the Http Pact interface (/send, etc.)(inprocess) API  in Chainweb
+module Chainweb.Test.Pact.RemotePactTest where
+
+import Control.Concurrent hiding (readMVar, putMVar)
+import Control.Concurrent.Async
+import Control.Concurrent.MVar.Strict
+import Control.Exception
+import Control.Lens
+import Control.Monad
+
+import qualified Data.Aeson as A
+import qualified Data.ByteString as BS
+import qualified Data.HashMap.Strict as HM
+import Data.Maybe
+import Data.Proxy
+import Data.Streaming.Network (HostPreference)
+import Data.String.Conv (toS)
+
+import Network.HTTP.Client.TLS as HTTP
+import Network.Connection as HTTP
+
+import Numeric.Natural
+
+import Servant.Client
+import System.FilePath
+import System.LogLevel
+import System.Time.Extra
+
+import Test.Tasty.HUnit
+import Test.Tasty
+import Test.Tasty.Golden
+
+import Text.RawString.QQ(r)
+
+import Pact.Types.API
+
+-- internal modules
+
+import Chainweb.Chainweb
+import Chainweb.Chainweb.PeerResources
+import Chainweb.Graph
+import Chainweb.HostAddress
+import Chainweb.Logger
+import Chainweb.Miner.Config
+import Chainweb.NodeId
+import Chainweb.Pact.RestAPI
+import Chainweb.Payload.PayloadStore (emptyInMemoryPayloadDb)
+import Chainweb.Test.P2P.Peer.BootstrapConfig
+import Chainweb.Test.Pact.Utils
+import Chainweb.Utils
+import Chainweb.Version
+
+import P2P.Node.Configuration
+import P2P.Peer
+
+apiSend :: SubmitBatch -> ClientM RequestKeys
+apiSend = client (Proxy :: Proxy SendApi)
+
+apiPoll :: Poll -> ClientM PollResponses
+apiPoll = client (Proxy :: Proxy PollApi)
+
+nNodes :: Natural
+nNodes = 1
+
+tests :: IO TestTree
+tests = do
+    peerInfoVar <- newEmptyMVar
+    let cwVersion = TestWithTime petersonChainGraph
+    theAsync <- async $ runTestNodes Warn cwVersion nNodes Nothing peerInfoVar
+    link theAsync
+    newPeerInfo <- readMVar peerInfoVar
+    let thePort = _hostAddressPort (_peerAddr newPeerInfo)
+
+    putStrLn $ "Server listening on port: " ++ show thePort ++ ". Sending client request..."
+
+    env <- getClientEnv thePort cwVersion
+    tts <- sendTest env
+    return $ testGroup "PactRemoteTest" tts
+
+sendTest :: ClientEnv -> IO [TestTree]
+sendTest env = do
+    let msb = A.decode $ toS escapedCmd :: Maybe SubmitBatch
```

Comment:
use `decodeStrictOrThrow` from Chainweb.Utils

Reply:
The test shuts down when a response is retrieved from poll, or after a fixed number of retries.  The time it takes running local is 2.1752 seconds with a 2 second wait before the poll.  Any lesser wait time actually makes it take longer due to poll failing, etc.

So everything else is inside the 0.1752 seconds...

Ok, I'll add the decodeStrictOrThrow and add the other stuff to another PR.